include/seqan3/align/alignment.hpp

```diff
@@ -0,0 +1,196 @@
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+// Author: Joerg Winkler <j.winkler AT fu-berlin.de>
+// ============================================================================
+
+#pragma once
+
+#include <iomanip>
+#include <iostream>
+#include <tuple>
+#include <type_traits>
+#include <utility>
+
+#include <range/v3/algorithm/for_each.hpp>
+#include <range/v3/iterator_range.hpp>
+#include <range/v3/view/all.hpp>
+#include <range/v3/view/slice.hpp>
+#include <range/v3/view/zip.hpp>
+
+#include <seqan3/alphabet.hpp>
+#include <seqan3/range/view/to_char.hpp>
+
+namespace seqan3
+{
+
+template <typename t>
+concept bool aligned_sequence_concept = true;
+
+//! Alignment class.
+/*!
+ * An alignment is a tuple of at least two aligned sequences.
+ * \tparam sequence_ts sequence types of the alignment
+ */
+template <typename ...sequence_ts>
+    requires (aligned_sequence_concept<sequence_ts> && ...)
+class alignment : public std::tuple<sequence_ts...>
+{
+public:
+    //! The alignment depth is the number of sequences contained.
+    static std::size_t const depth = sizeof...(sequence_ts);
+
+    //! Constructor that allows to pass sequences directly.
+    constexpr alignment(sequence_ts && ...sequences)
+        : std::tuple<sequence_ts...>(std::forward<sequence_ts>(sequences)...)
+    {
+        static_assert(sizeof...(sequence_ts) > 1, "An alignment requires at least two sequences.");
+    };
+};
+
+/*!
+ * Alignment class template deduction.
+ * \tparam sequence_ts sequence types of the alignment
+ * \sa http://en.cppreference.com/w/cpp/language/class_template_deduction
+ */
+template <typename ...sequence_ts>
+alignment(sequence_ts...) -> alignment<sequence_ts...>;
+
+//! Type for column_iterator.
+template <typename ...sequence_ts>
+using column_iterator_type = ranges::v3::zip_view<ranges::v3::iterator_range<typename sequence_ts::const_iterator,
+                                                                             typename sequence_ts::const_iterator>...>;
+/*!
+ * Column-wise iteration over a sequence alignment.
+ * \tparam sequence_ts sequence types of the alignment
+ * \param align the underlying alignment for the iterator
+ * \return a column iterator for the given alignment
```

Comment:
Please add a code example since the usage of the column iterator is not trivial.